framework/elements/BaseElement.js

```diff
@@ -0,0 +1,107 @@
+import { Timeouts } from '../../constants/timeouts'
+
+class BaseElement {
+  constructor(name, selector) {
+    this.name = name
+    this.selector = selector
+  }
+
+  async getElement() {
+    return await $(this.selector)
```

Comment:
await is redundant here
also it would be better for avoiding potential issues to get this element on a separate  line with await and use it further in methods, e.g.:
const element = await this.getElement()
return element.getText()